x/evm/migrations/v4/migrate.go

```diff
@@ -0,0 +1,56 @@
+package v4
+
+import (
+	"github.com/cosmos/cosmos-sdk/codec"
+	storetypes "github.com/cosmos/cosmos-sdk/store/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	v4types "github.com/evmos/ethermint/x/evm/migrations/v4/types"
+	"github.com/evmos/ethermint/x/evm/types"
+)
+
+// MigrateStore migrates the x/evm module state from the consensus version 3 to
+// version 4. Specifically, it takes the parameters that are currently stored
+// and managed by the Cosmos SDK params module and stores them directly into the x/evm module state.
+func MigrateStore(
+	ctx sdk.Context,
+	storeKey storetypes.StoreKey,
+	legacySubspace types.Subspace,
+	cdc codec.BinaryCodec,
+) error {
+	store := ctx.KVStore(storeKey)
+	var params v4types.Params
+	legacySubspace.GetParamSet(ctx, &params)
```

Comment:
we should use the following logic `GetIfExists` so that we prevent panics


```go
for _, pair := range params.ParamSetPairs() {
		k.paramSpace.GetIfExists(ctx, pair.Key, pair.Value)
}
```

Reply:
@fedekunze there is a more convenient function in the Subspace `GetParamSetIfExists`. Would it suffice so I avoid adding multiple functions to the legacy `Subspace` interface?

```go
// GetParamSetIfExists iterates through each ParamSetPair where for each pair, it will
// retrieve the value and set it to the corresponding value pointer provided
// in the ParamSetPair by calling Subspace#GetIfExists.
func (s Subspace) GetParamSetIfExists(ctx sdk.Context, ps ParamSet) {
	for _, pair := range ps.ParamSetPairs() {
		s.GetIfExists(ctx, pair.Key, pair.Value)
	}
}

```